### PR TITLE
Add tests to increase spec coverage

### PR DESCRIPTION
**Pull request by fb55**
parse5 is now tracking the test coverage of its codebase. The tests here all cover areas that previously didn't have tests.

### CODE REVIEW

File: tree-construction/doctype01.dat [hunk 1]
```diff
@@ -368,14 +368,17 @@
 
 #data
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Frameset//EN"
-"http://www.w3.org/TR/xhtml1/DTD/xhtml1-frameset.dtd">
```
Comment (fb55) [inline tree-construction/doctype01.dat:371]: There was no test for the `<p><table>` behavior in limited quirks mode.

File: tree-construction/doctype01.dat [hunk 2]
```diff
@@ -472,3 +475,43 @@
 | <html>
 |   <head>
 |   <body>
+
+#data
+<!DOCTYPE html SYSTEM "http://www.ibm.com/data/dtd/v11/ibmxhtml1-transitional.dtd"><p><table>
```
Comment (fb55) [inline tree-construction/doctype01.dat:480]: All of these are special cases for quirks mode that were not covered. The `<p><table>` at the end ensures we do end up in quirks mode.

File: tree-construction/doctype01.dat (hunk 1 above)
Comment (gsnedders) [reply tree-construction/doctype01.dat:371]: Can you add this as a new test, not replacing the existing one?

**Review by gsnedders — COMMENTED** [review]
sorry for not seeing this before (yikes)!

mostly, I think we should just move the quirks-mode tests into a new file, rather than scattering them around

File: tree-construction/doctype01.dat (hunk 1 above)
Comment (fb55) [reply tree-construction/doctype01.dat]: Done!

**Comment by fb55** [comment]
Thanks for the review @gsnedders, made the requested changes. I would like to wait for the open PRs for #141 to be landed before this one, to make sure I didn't mess up subtleties around errors.

**Review by gsnedders — APPROVED** [review]

**Comment by gsnedders** [comment]
(Let me know if you do indeed think this is ready to land.)

File: tree-construction/tables01.dat [hunk 3]
```diff
@@ -306,3 +306,15 @@
 |       <table>
 |       <s>
 |       <table>
+
+#data
+<table>a<!doctype html>
+#errors
+(1,7): expected-doctype-but-got-start-tag
+(1,30): expected-closing-tag-but-got-eof
+#document
```
Comment (Ms2ger) [inline tree-construction/tables01.dat:315]: nokogiri thinks this should have four errors

**Comment by fb55** [comment]
This should now be ready to land.

**Pull request merged**